student-work/2022/Jonathan-Kemp/api/GET.md

```diff
@@ -0,0 +1,72 @@
+# GET /meal
+
+
+# HTTP Responses 
+
+| Status          | Type     | Code Description                                      |
+|----------------|----------|--------------------------------------------------|
+| 200 - OK    | string  | Order is correct. |
+| 400 - Bad Request | string | Meal ordered is incorrect |
+
+## GET Retrieve the currently available meal 
+Retrieves the meal that is available at the current time. 
+## GET /meal 
+Arguments (Query Parameters) 
+There are no arguments. 
+
+### Response Parameters
+
+| Parameter name | Data type   | Required? | Description                          |
+|------------------|--------|-----------|--------------------------------------|
+| Patty | string | Required  | "largePatty" - for a large patty "mediumPatty" - for a medium patty "smallPatty" - for a small patty|
+| r | string |  Required  |  Cooked rare |
+| mr | string |  Required  |  Cooked medium rare |
+| m | string |  Required  |  Cooked medium |
+| wd | string |  Required  |  Cooked well done |
+| Bun | string |  Required  |  "seededBun" - for a seeded bun "regularBun" - for a regular bun "chiabattaBun" - for a chiabatta bun "glutenFreeBun" - for a gluten free bun|
+| filling1| string |  Required  | Lettuce filling |
+| filling2 | string |  Required  | Tomato filling |
+| filling3 | string |  Required  |  Pickle filling |
+| fries | string |  Required  |  Fries side |
+| curlyFries | string |  Required  | Curly fries side |
+| sweetPotatoFries  | string |  Required  |  Sweet potato fries side |
+| cola  | string |  Required  | Cola drink |
+| lemonade  | string |  Required  |  Lemonade drink |
+| rootBeer | string |  Required  |  Root beer drink |
+        
```

Comment:
```
| Parameter name | Data type   | Required? | Description                          |
|------------------|--------|-----------|--------------------------------------|
| PattySize | string | Required  | "L", "M" or "S"|
| BurgerCook| string |  Required  |  "R", "MR", "M", or "WD" |
| BunType | string |  Required  |  "Seeded", "Plain", "ChiabattaBun", or "glutenFreeBun"|
| filling1| string |  Required  | "Lettuce", "Tomato", or "Pickle"|
| filling2 | string |  Required  | "Lettuce", "Tomato", or "Pickle" |
| filling3 | string |  Required  |  "Lettuce", "Tomato", or "Pickle"|
| FriesType| string |  Required  |  "RegularFries", "curlyFries", or "SweetPotatoFries" |
| DrinkType| string |  Required  | "Cola", "Lemonade", or "RootBeer"|       
```

Reply:
Use a table like this one. 
Notice how it is more compact
and makes more sense as it tells the developer all the options for the parameters.

For example "PattySize":"M"
                     "BurgerCook": "MR"
                     "BunType":"Seeded"
                     "Fixing1":"Lettuce"
                    "FriesType":"SweetPotatoFries"
                    "DrinkType":"Lemonade"

Will give me a medium burger cooked medium rare on a sesame bun with sweet potato fries and a lemonade.

Reply:
use the same fix for all tables.